mobilecoind-dev-faucet/src/bin/main.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+#![deny(missing_docs)]
+
+//! HTTP faucet service backed by mobilecoind
+
+#![feature(proc_macro_hygiene, decl_macro)]
+
+use clap::Parser;
+use mc_common::logger::{create_app_logger, log, o};
+use mc_mobilecoind_dev_faucet::{data_types::*, Config, State};
+use rocket::{get, post, routes, serde::json::Json};
+
+/// Request payment from the faucet, and map the rust result onto json for
+/// rocket appropriately
+#[post("/", format = "json", data = "<req>")]
+async fn post(
+    state: &rocket::State<State>,
+    req: Json<JsonFaucetRequest>,
+) -> Json<JsonSubmitTxResponse> {
+    Json(match state.handle_post(&req).await {
+        Ok(resp) => resp.into(),
+        Err(err_str) => JsonSubmitTxResponse {
+            success: false,
+            err_str: Some(err_str),
+            ..Default::default()
+        },
+    })
```

Comment:
Optional/Equivalent: Consider moving this error handling into an `impl From<Result<mc_mobilecoind_api::SubmitTxResponse, String> for JsonSubmitTxResponse`, then this method can be
```suggestion
    Json(state.handle_post(&req).await.into())
```